### ISSUE TITLE
Continuing to get Xcode crash on `CMD+S` save with SwiftFormat

### ISSUE THREAD

**Issue by WestFlow127** [nicklockwood/SwiftFormat#1897]
I have recently upgraded to macOS Sonoma/Xcode 16, and continue to get an Xcode crash when I save my swift file.
It does not happen every time, but about ~10% of file-saves do this.
My current `SwiftFormat` build is `0.54.5`.

Would it help if I attached my Xcode crash report here?
I thoroughly like using SwiftFormat, but the Xcode crashes can cause me to lose changes.
Thanks

**Comment by nicklockwood** [comment]
I'm a bit confused. SwiftFormat does not have a mechanism for formatting on save in Xcode - is that something you have implemented yourself?

And if not, what leads you to believe that SwiftFormat is the cause of the crashes?

**Comment by WestFlow127** [comment]
@nicklockwood yes, I've added a keyboard shortcut demonstrated in this tutorial [here](https://medium.com/@jozott/format-on-save-xcode-swift-8133d049b3ac).
The indication of something SwiftFormat related with regards to the crash is the crash happens immediately after the shortcut `CMD+S` and otherwise does not occur.

**Comment by WestFlow127** [comment]
In theory, it may be because Xcode is trying to save the file during the formatting process (writing to the swift files) which bugs out Xcode to abort.
